<launch>
  <arg name="image_width" default="640"/>
  <arg name="image_height" default="480"/>
  <arg name="viewer_ok" default="true"/>
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="$(arg image_width)" />
    <param name="image_height" value="$(arg image_height)" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
  </node>
  <node name="image_view" pkg="image_view" type="image_view" if="$(arg viewer_ok)" respawn="false" output="screen">
    <remap from="image" to="/usb_cam/image_raw"/>
    <param name="autosize" value="true" />
  </node>
</launch>
